### PR TITLE
timer: add sys table `mysql.tidb_timers` to store all timers in tidb

### LINKED ISSUE #45128: Add system table `mysql.tidb_timers` to manage all timers in tidb

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #45128

### What is changed and how it works?

A system table `mysql.tidb_timers` is added in this PR. It is used for store all timers in tidb in the future.

The TTL feature will use this table to manage its own TTL job firstly

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
timer: add sys table `mysql.tidb_timers` to store all timers in tidb
```

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -6900,7 +6900,6 @@ func BuildHiddenColumnInfo(ctx sessionctx.Context, indexPartSpecifications []*as
 				return nil, errors.Trace(err)
 			}
 		}
-		idxPart.Expr = nil
```
Comment (lcwangchao) [inline ddl/ddl_api.go:6903]: We should remove this line to keep `IndexPartSpecification` immutable because here: https://github.com/pingcap/tidb/blob/96dcbb7c9a20f417fe514d0c0a21abfae553f0d7/ddl/schematracker/checker.go#L220-L232 will call `CreateTable` twice.

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -7033,6 +7032,14 @@ func (d *ddl) createIndex(ctx sessionctx.Context, ti ast.Ident, keyType ast.Inde
 		return d.addHypoIndexIntoCtx(ctx, ti.Schema, ti.Name, indexInfo)
 	}
 
+	// We need to set `IndexPartSpecification.Expr` because json cannot unmarshal it correctly.
+	indexPartSpecificationsArg := make([]*ast.IndexPartSpecification, len(indexPartSpecifications))
```
Comment (bb7133) [inline ddl/ddl_api.go:7036]: Is this change related to this PR?

Comment (lcwangchao) [reply ddl/ddl_api.go:7036]: This PR will create a system table with multi-value index. If this change not included, some tests will fail

File: ddl/ddl_api.go (hunk 1 above)
Comment (bb7133) [reply ddl/ddl_api.go:6903]: I wonder if there's any bug when this is not removed?

File: ddl/ddl_api.go (hunk 2 above)
Comment (bb7133) [reply ddl/ddl_api.go:7036]: Got it.

**Comment by bb7133** [comment]
The 'br-integration-test' failed, PTAL

File: ddl/ddl_api.go (hunk 1 above)
Comment (lcwangchao) [reply ddl/ddl_api.go:6903]: I'm not sure whether we should call it bug, beacause it only happens in test cases.

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by YangKeao — APPROVED** [review]
LGTM
